### ISSUE TITLE
Reduce complexity of `view_list` in views.py

### ISSUE THREAD

**Issue by bittner** [shacker/django-todo#28]
This is not a serious bug, but a QA topic. `flake8` complains that [view_list()](https://github.com/shacker/django-todo/blob/master/todo/views.py#L86) is too complex:
```
./todo/views.py:85:1: C901 'view_list' is too complex (21)
```
The function needs some heavy refactoring. Ideally, cyclomatic complexity [should not exceed 10](https://en.wikipedia.org/wiki/Cyclomatic_complexity#cite_ref-mccabe76_3-3).

**Comment by shacker** [comment]
I agree this function is getting kind of ridiculous, but am not seeing the flake8 warning against this file, either in Atom with the flake8 linter, or with the flake8 CLI. How can I reproduce this?

**Comment by bittner** [comment]
I've set `max-complexity` to 21 [in tox.ini](https://github.com/shacker/django-todo/blob/master/tox.ini#L23) to make `flake8` pass. If you set it back to 10 you get the complaint.

BTW, all you need to do is run `python setup.py test` for flake8 to run. No need to install anything (apart from `virtualenv`) as mentioned in the README.

**Comment by shacker** [comment]
OK, I see it now. The view does a lot, and the function for the view reflects that complexity. Is this a case where the linter might be encouraging us to abstract out things that really don't warrant abstracting? I've pushed some small optimizations, but looking at it, I'm having trouble seeing how things could be more DRY.  Ideas? And is this a 1.6 blocker or not?

**Comment by bittner** [comment]
I think what cyclomatic complexity is trying to tell us is that there are more `if`'s and `elif`'s that you can easily understand (without sitting in front of the code rather long, with the potential danger to overlook some serious issue). Code should be simple and straight-forward. The easiest way to do this is split long functions up into several ones, and give each sub-function a self-explanatory name.

When I look at the `if`-clauses like the following my first impression is, that should be be views of their own. They all have their own business function I guess, they all (could) have their own URL path / scheme. Or maybe there is some other, elegant way to implement what all this complex concatenation of logic does:
``` python
if request.POST.getlist('mark_done'):
if request.POST.getlist('undo_completed_task'):
if request.POST.getlist('del_completed_task'):
elif list_slug == "recent-add":
elif list_slug == "recent-complete":
if request.POST.getlist('add_task'):
```
Of course, I'm just looking at the code superficially. Actually, that's another sign: I should be able to look at the code and "immediately" understand what it does. And I don't.

And no, this is not a 1.6 blocker. It's a code smell that may be good to address some time.

**Comment by shacker** [comment]
Yeah, I'll have to sit down with that one day for a closer look. Meanwhile, the version has bumped to 1.6 and we're ready to go. Thanks again.

**Comment by shacker** [comment]
Did a lot of work tonight to fix some broken tox tests and refactored `view_list` to bring complexity down to 10. There's probably more work to be done but this feels better already.

**Comment by shacker** [comment]
This is fixed in the upcoming version 2. Sorry for the long delay!
